apis/apps/v1alpha1/nebulacluster_common.go

```diff
@@ -214,10 +214,12 @@ func generateAgentContainer(c NebulaClusterComponentter, init bool) corev1.Conta
 		}
 	}
 
+	agentImage := getImage(c.GetNebulaCluster().Spec.Agent.Image, c.GetNebulaCluster().Spec.Agent.Version, defaultMetadImage)
 	container := corev1.Container{
-		Name:    AgentSidecarContainerName,
-		Image:   defaultAgentImage,
-		Command: agentCmd,
+		Name:      AgentSidecarContainerName,
+		Image:     agentImage,
+		Command:   agentCmd,
+		Resources: c.GetNebulaCluster().Spec.Agent.Resources,
```

Comment:
Please rebase to master branch, here has a commit conflict.

Reply:
have already rebase

Reply:
The function 'getImage' has been removed. And I see you merged my previous PR.

Reply:
i've delete getImage function call,but i got some error on check
```
[root@base01 nebula-operator]# make check
go mod tidy
# go fmt ./...
/opt/goworkspace/bin/gofumpt -w -l ./
go vet ./...
# github.com/vesoft-inc/nebula-operator/pkg/ngctl/cmd/info
pkg/ngctl/cmd/info/clusterinfo.go:97:39: nc.GraphdComponent().GetImage undefined (type "github.com/vesoft-inc/nebula-operator/apis/apps/v1alpha1".NebulaClusterComponent has no field or method GetImage)
pkg/ngctl/cmd/info/clusterinfo.go:246:65: undefined: v1alpha1.NebulaClusterComponentter
# github.com/vesoft-inc/nebula-operator/pkg/scheduler/extender/predicates
pkg/scheduler/extender/predicates/topology.go:85:24: component.GetReplicas undefined (type v1alpha1.NebulaClusterComponent has no field or method GetReplicas)
pkg/scheduler/extender/predicates/topology.go:125:50: undefined: v1alpha1.NebulaClusterComponentter
# github.com/vesoft-inc/nebula-operator/pkg/controller/nebulacluster
vet: pkg/controller/nebulacluster/nebula_cluster_condition_updater_test.go:78:25: PodSpec not declared by package v1alpha1
# github.com/vesoft-inc/nebula-operator/tests/e2e/nebulacluster
tests/e2e/nebulacluster/util.go:60:5: unknown field 'PodSpec' in struct literal of type "github.com/vesoft-inc/nebula-operator/apis/apps/v1alpha1".GraphdSpec
tests/e2e/nebulacluster/util.go:60:23: undefined: v1alpha1.PodSpec
tests/e2e/nebulacluster/util.go:75:5: unknown field 'PodSpec' in struct literal of type "github.com/vesoft-inc/nebula-operator/apis/apps/v1alpha1".MetadSpec
tests/e2e/nebulacluster/util.go:75:23: undefined: v1alpha1.PodSpec
tests/e2e/nebulacluster/util.go:98:5: unknown field 'PodSpec' in struct literal of type "github.com/vesoft-inc/nebula-operator/apis/apps/v1alpha1".StoragedSpec
tests/e2e/nebulacluster/util.go:98:23: undefined: v1alpha1.PodSpec
make: *** [vet] Error 1
```

Reply:
my fault, ignore it

Reply:
> i've delete getImage function call,but i got some error on check
> 
> ```
> [root@base01 nebula-operator]# make check
> go mod tidy
> # go fmt ./...
> /opt/goworkspace/bin/gofumpt -w -l ./
> go vet ./...
> # github.com/vesoft-inc/nebula-operator/pkg/ngctl/cmd/info
> pkg/ngctl/cmd/info/clusterinfo.go:97:39: nc.GraphdComponent().GetImage undefined (type "github.com/vesoft-inc/nebula-operator/apis/apps/v1alpha1".NebulaClusterComponent has no field or method GetImage)
> pkg/ngctl/cmd/info/clusterinfo.go:246:65: undefined: v1alpha1.NebulaClusterComponentter
> # github.com/vesoft-inc/nebula-operator/pkg/scheduler/extender/predicates
> pkg/scheduler/extender/predicates/topology.go:85:24: component.GetReplicas undefined (type v1alpha1.NebulaClusterComponent has no field or method GetReplicas)
> pkg/scheduler/extender/predicates/topology.go:125:50: undefined: v1alpha1.NebulaClusterComponentter
> # github.com/vesoft-inc/nebula-operator/pkg/controller/nebulacluster
> vet: pkg/controller/nebulacluster/nebula_cluster_condition_updater_test.go:78:25: PodSpec not declared by package v1alpha1
> # github.com/vesoft-inc/nebula-operator/tests/e2e/nebulacluster
> tests/e2e/nebulacluster/util.go:60:5: unknown field 'PodSpec' in struct literal of type "github.com/vesoft-inc/nebula-operator/apis/apps/v1alpha1".GraphdSpec
> tests/e2e/nebulacluster/util.go:60:23: undefined: v1alpha1.PodSpec
> tests/e2e/nebulacluster/util.go:75:5: unknown field 'PodSpec' in struct literal of type "github.com/vesoft-inc/nebula-operator/apis/apps/v1alpha1".MetadSpec
> tests/e2e/nebulacluster/util.go:75:23: undefined: v1alpha1.PodSpec
> tests/e2e/nebulacluster/util.go:98:5: unknown field 'PodSpec' in struct literal of type "github.com/vesoft-inc/nebula-operator/apis/apps/v1alpha1".StoragedSpec
> tests/e2e/nebulacluster/util.go:98:23: undefined: v1alpha1.PodSpec
> make: *** [vet] Error 1
> ```

my fault, ignore it